core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerBackend.scala

```diff
@@ -124,6 +124,13 @@ private[spark] class MesosSchedulerBackend(
         Value.Scalar.newBuilder()
           .setValue(MemoryUtils.calculateTotalMemory(sc)).build())
       .build()
+
+    sc.conf.getOption("spark.mesos.uris").map { uris =>
+      uris.split(",").foreach { uri =>
+        command.addUris(CommandInfo.URI.newBuilder().setValue(uri.trim()))
+      }
+    }
```

Comment:
There's a common superclass of all these three classes called `MesosSchedulerUtils`. I think this code could go there.